### DISCUSSION TITLE
file validation with yup does not work well

### DISCUSSION THREAD

**Discussion by 55enokky** [react-hook-form/react-hook-form#2576] [Q&A]
hi,
I got problem with validating my input file with yup. 
So far I got my validation schema written like this:
```typesctipt
const schema = yup.object().shape({
  file1: yup.mixed().test("type", "We only support jpeg", (value) => {
    return value && value[0].type === "image/jpeg";
  }),
  file2: yup.mixed().test("type", "We only support jpeg", (value) => {
    return value && value[0].type === "image/jpeg";
  })
});
```
No error will be displayed on the first selection, and the second and subsequent selections will behave correctly.

https://codesandbox.io/s/confident-dawn-8hv96

Did i do anything wrong ?

**Comment by bluebill1049** [comment]
<img width="1650" alt="Screen Shot 2020-08-13 at 11 37 56 am" src="https://user-images.githubusercontent.com/10513364/90084721-71d78900-dd59-11ea-8fe4-9a102ac38d70.png">

https://codesandbox.io/s/pedantic-wright-2zp5v?file=/src/App.js

we expected `type` from the validation, looks like it's missing and it throws a type error as well.

**Reply by 55enokky** [reply]
Thanks for the reply!!

I changed schema to the following.
```typescript
file1: yup
    .mixed()
    .test(
      "type",
      "We only support jpeg",
      (value) => !value || (value && value[0].type === "image/jpeg")
    )
```
This validation works fine as we would expect. 
https://codesandbox.io/s/agitated-sinoussi-gv6ci?file=/src/App.js
![image](https://user-images.githubusercontent.com/4176481/90327539-6db4a100-dfcf-11ea-8e00-501fb8278072.png)

However, if the schema has two same validations, the type error will be returned as before.
https://codesandbox.io/s/festive-banzai-tkiuj?file=/src/App.js
![image](https://user-images.githubusercontent.com/4176481/90327559-9fc60300-dfcf-11ea-8100-fffe2b719fe3.png)

What's wrong with my notation ?

**Reply by bluebill1049** [reply]
Thanks for the following up, we have a bug report and I think it's related: https://github.com/react-hook-form/resolvers/issues/43
